Makefile

```diff
@@ -16,12 +28,17 @@ ifeq ($(COMPILER),clang)
 else
 	$(MAKE) -C src/vdp
 endif
+
+ifeq ($(OS_DETECTED),Windows)
+	copy src\vdp\*.so firmware
+else
 	cp src/vdp/*.so firmware/
+endif
 
 cargo:
-ifeq ($(OS),Windows_NT)
+ifeq ($(OS_DETECTED),Windows)
 	set FORCE=1 && cargo build -r
-	cp ./target/release/fab-agon-emulator.exe .
+	copy target\release\fab-agon-emulator.exe .
```

Comment:
There are also other makefiles in emulator's source tree (for example, in VDP sources) which may also use *nix specific commands, and in case they do, will need to be fixed in a similar way.

Reply:
There is a difference. Make can handle those unix paths because its a GNU tool. So when you call make on windows it works. But the windows tools like copy can't handle them.

Reply:
Yes, I meant not file paths but commands like 'cp', 'rm', etc.
For example, the makefile under fab-agon-emulator\src\vdp\userspace-vdp-gl\src\ contains some 'rm' commands.

Reply:
Oh yes, thats a real problem. Make clean doesn't fully work right now.

Reply:
I think 'make clean' does work now, assuming a compatibility layer like coreutils is used to provide implementation of missing Linux commands under Windows.

I'm wondering if it is worth the efforts at some point to switch to a higher-level build system like CMake, which in theory could allow us to get rid of the platform-specific make files.

Reply:
Im pretty sure cargo can do this. You can write arbitrary rust code in build.rs to move around files operating system agnostic. My first attempt was to remove make for everything and rely on rust cc crate to build the vdp, but this can only create static libs. Make already is an extra tool which feels weird in a rust context, but it doesnt look like cargo can easily build the shared libs.